### PR TITLE
Ability to update profile image

### PR DESCRIPTION
**Pull request by laredotornado**
This adds a component and the ability to update the user's profile image (avatar).

Design is here -- https://www.figma.com/file/v5HdxDsvO2NzMuZU6CArzF/Resilience-App?node-id=1238%3A2

Dev Story is here -- https://airtable.com/tblRjd2McKgRW9MKj/viwlRRNI0PZIxrGkc/recRjjPmPj91aQRY5?blocks=hide

### CODE REVIEW

**Comment by jwu910** [comment]
Hi @laredotornado Seems like an awesome component to create. Just want to ask if it would be best to build out the entire functionality for photo cropping if there are react libraries that seem to already do this? 

I think its a question of complexity and maintenance. Can we achieve this ticket by using something like [react-easy-crop](https://github.com/ricardo-ch/react-easy-crop) or [react-image-crop](https://github.com/DominicTobias/react-image-crop) and just build our own profile image component around one of these?

**Comment by laredotornado** [comment]
> Hi @laredotornado Seems like an awesome component to create. Just want to ask if it would be best to build out the entire functionality for photo cropping if there are react libraries that seem to already do this?
> 
> I think its a question of complexity and maintenance. Can we achieve this ticket by using something like [react-easy-crop](https://github.com/ricardo-ch/react-easy-crop) or [react-image-

@jwu910 , K, I removed the home-grown stuff with the [library you suggested](https://github.com/DominicTobias/react-image-crop)

File: src/app/page/UserProfile/ProfileImage.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const firebaseProfile = useSelector((state) => state.firebase.profile);
+
+  const displayName = _.get(profile, "displayName", "");
+  const profilePhoto = _.get(profile, "photoURL", "");
+
+  var [defaultPreview, setDefaultPreview] = useState("defaultPreview");
+  var [view, setView] = useState("view");
+
+  function onCropDefault(preview) {
+    console.log("on crop default called.");
```
[the remark this answers is not in the document]
Comment (laredotornado) [reply src/app/page/UserProfile/ProfileImage.jsx]: Removed the debugging and made some changes, which I "think" will fix the crop issue.  I won't resolve these conversations until you approve, but I can if that's the protocol folks are using here.

File: src/app/page/UserProfile/ProfileImage.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @param {func} setProfile - A function that can set the profile object. 
+ * @return {Component} the component for displaying and editing the profile image
+ */
+const ProfileImage = ({ classes, profile, setProfile }) => {
+  const firebase = useFirebase();
+  const firestore = useFirestore();
+  const firebaseProfile = useSelector((state) => state.firebase.profile);
+
+  const displayName = _.get(profile, "displayName", "");
+  const profilePhoto = _.get(profile, "photoURL", "");
```
[the remark this answers is not in the document]
Comment (laredotornado) [reply src/app/page/UserProfile/ProfileImage.jsx]: Done!

File: src/app/page/UserProfile/ProfileImage.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  var [defaultPreview, setDefaultPreview] = useState("defaultPreview");
+  var [view, setView] = useState("view");
```
[the remark this answers is not in the document]
Comment (laredotornado) [reply src/app/page/UserProfile/ProfileImage.jsx]: Done!

**Comment by mat10tng** [comment]
seem like build are failing

File: src/app/page/UserProfile/photoselection.jsx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    return this.state.imgHeight
+  }
+
+  get halfHeight() {
+    return this.state.imgHeight / 2
+  }
+
+  get image() {
+    return this.state.image
+  }
```
[the remark this answers is not in the document]
Comment (laredotornado) [reply src/app/page/UserProfile/photoselection.jsx]: I eliminated all the "get" methods except the one for loaderId.  I was having some trouble getting the files to load when I removed that and replaced its invocation with the function contents.  However, I can work on replacing that too if this is not a pattern the group uses.

File: src/app/page/UserProfile/photoselection.jsx [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+            </div>
+            : 
+              <ReactCrop style={style} src={this.image.src} crop={this.state.crop} onChange={newCrop => this.onCrop(newCrop)} circularCrop="true" />
+        }
+      </div>
+    )
+  }
+}
+
+export default PhotoSelection
```
[the remark this answers is not in the document]
Comment (laredotornado) [reply src/app/page/UserProfile/photoselection.jsx:184]: Whenever you have time, I tried to cut a bunch of unnecessary code.  Let me know if it makes more sense or if you have suggestions for restructuring.  Thanks!

**Pull request merged**